### PR TITLE
Add game logic

### PR DESCRIPTION
**Pull request by KabohaJeanMark**
Refactor code to add OOP classes for the player and board to encapsulate their operations.
Add helper methods to implement the tic tac toe game logic in a DRY manner.
Implement game logic through data structures hashes and a multidimensional array creating algorithms to check for possible winning combinations and to validate the input clearly displaying instructions to the users.
Update the README.md with a visual walkthrough of how the game works.

### CODE REVIEW

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hey @KabohaJeanMark and @chasscepts,
I'm happy to see how good the project is, Your PR title and summary are great, The README file is super, the game validations are great.
I like the way the moves are implemented.

One thing has prevented you guys from moving forward, let's fix it and move😂🤣😎😪✔
**Move board and player files into lib.
Remove all commented code.**
When you are done with the changes, submit for another review🎉🎉🎉👔🎩🎓✔✔

 [OPTIONAL] reduce the sleep time, it is taking long, I almost thought it has stopped.

If you have any question, please find me on slack and twitter@Elisha1k
Cheers.

**Comment by chasscepts** [comment]
@elisha2kyakpo1 thanks for the review. We have made all the changes requested in the feedback.
- We moved board and player files into the lib folder.
- We removed comments from the code
- We reduced the game transition delays

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  attr_reader :name, :token
+
+  def initialize(name, token)
+    @name = name.capitalize
+    @token = token
+  end
+
+  def to_s
+    @name
+  end
```
Comment (henatan99) [inline lib/player.rb]: This getter method is not required as you have the 'attr_reader  :name, :token'  getter at the first line of Player class. Make your code DRY. To make corresponding changes to this one, you can use the attributes directly in Player objects. 
eg:- player.name for the player's name and player.token for the player's token

File: bin/main.rb [hunk 2]
```diff
@@ -1,42 +1,30 @@
 #!/usr/bin/env ruby
 
-CHARACTER_1 = 'X'.freeze
-CHARACTER_2 = 'O'.freeze
+require_relative './lib/board'
+require_relative './lib/player'
```
Comment (henatan99) [inline bin/main.rb]: These file paths are wrong. Kindly, provide the correct files' paths and make test runs for yourself.

**Review by henatan99 — CHANGES_REQUESTED** [review]
# Changes Requested :recycle: 
Hi @KabohaJeanMark, great work so far, :+1: but there are things you need to fix before your project gets approved.

## To Highlight 
- Your game works exactly as intended 
- Nice Readme, descriptive PR summay and commit messages

## To Improve 
- The file paths in the main.rb file is wrong. Please, check the inline comment and fix it.
- You have used unnecessary getter method(to_s) in your Player class, yet you have used a shorthand method attr_reader already. Do not Repeat Yourself! Doing so, you can use the name attribute whenever you want the name of a player object.

### Kindly fix these issues and submit again for re-review.

If you want to contact me, here are my links:
- [LinkedIn](https://www.linkedin.com/in/henok-mekonnen-2a251613/)
- [Twitter](https://twitter.com/henatan99)
- or Henok Mekonnen via Slack

**Comment by chasscepts** [comment]
@henatan99 thanks for your feedback. We have implemented all the changes requested.
- We corrected the path error in our require statement in game.rb file
- We refactored to remove the to_s method in the Player class
Please kindly review the changes to the project.

**Review by MkrtichSargsyan — APPROVED** [review]
## _Status:_ Approved  :hamster: :rocket:
### Hi Guys.

Congratulations on your hard work, your project is now approved, thank you for making the changes that previous reviewers suggested.

You can go ahead and merge this branch to the development branch of this repo.

Good luck in your future projects and happy coding :relaxed:

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

**Pull request merged**